client/src/app/shared/forms/form-validators/video-channel-validators.service.ts

```diff
@@ -14,28 +14,28 @@ export class VideoChannelValidatorsService {
     this.VIDEO_CHANNEL_NAME = {
       VALIDATORS: [
         Validators.required,
-        Validators.minLength(3),
-        Validators.maxLength(20),
-        Validators.pattern(/^[a-z0-9._]+$/)
+        Validators.minLength(1),
+        Validators.maxLength(50),
+        Validators.pattern(/^[a-z0-9][a-z0-9-._]*$/)
```

Comment:
`-` char is not accepted by mastodon i think

Reply:
I'll have to fix that, or maybe we can have mastadon accept it?

Reply:
There is two issues at least:
- https://github.com/tootsuite/mastodon/issues/9309
- https://github.com/tootsuite/mastodon/issues/9066

but that doesn't seems to be planned for the moment..

Reply:
To me it seems that mastodon accepts everything accept the `.` char, which it removes if it sees it.

https://github.com/tootsuite/mastodon/blob/master/app/validators/unique_username_validator.rb#L7

Reply:
<img width="286" alt="screen shot 2018-11-25 at 6 04 44 am" src="https://user-images.githubusercontent.com/43896676/48978867-15f63980-f078-11e8-9f55-ca4cd465fc30.png">

Reply:
I'm not sure this is the only thing.
According to Gargron here: https://github.com/tootsuite/mastodon/issues/9323#issuecomment-441005348 dashes are still refused from remote usernames.

Reply:
<img width="280" alt="screen shot 2018-11-25 at 6 06 07 am" src="https://user-images.githubusercontent.com/43896676/48978879-3e7e3380-f078-11e8-9825-899587ba4bc1.png">

Reply:
I'd say it's safe to merge the pull request without any other changes to it

Reply:
This is still not safe, see my previous comment.
You are only showing basic HTML5 validation (field filled or not).
Validate the form and it will refuse it:
![image](https://user-images.githubusercontent.com/8625/48978902-5f17af00-f0b3-11e8-8e0b-6aa9bf143d82.png)

There is two issues still opened, and Gargron stated three days ago that dashes are **not accepted** from remote usernames.

Reply:
Here's a scenario where a user can have spaces in their very long username, even beyond the 50 chars limit we have with this pull request. `very-long username-a user likes just because he likes it`
<img width="277" alt="screen shot 2018-11-25 at 6 14 14 am" src="https://user-images.githubusercontent.com/43896676/48978942-65893500-f079-11e8-9f84-60520662b63c.png">

Reply:
No, you didn't **submitted** the form, the HTML5 shows a green border only because the field is **filled** and not empty (because of the "required" html attribute).